core/lib/src/router/router.rs

```diff
@@ -55,6 +57,33 @@ impl Router {
             .flat_map(move |routes| routes.iter().filter(move |r| r.matches(req)))
     }
 
+    pub(crate) fn matches_except_formats<'r, 'a: 'r>(
+        &'a self,
+        req: &'r Request<'r>
+    ) -> bool {
+        self.routes.get(&req.method())
+            .into_iter()
+            .flatten()
+            .any(|route| paths_match(route, req) && queries_match(route, req))
+    }
+
+    const ALL_METHODS: &'static [Method] = &[
+        Method::Get, Method::Put, Method::Post, Method::Delete, Method::Options,
+        Method::Head, Method::Trace, Method::Connect, Method::Patch,
+    ];
+
+    pub(crate) fn matches_except_method<'r, 'a: 'r>(
+        &'a self,
+        req: &'r Request<'r>
+    ) -> bool {
+        Self::ALL_METHODS
+            .iter()
+            .filter(|method| *method != &req.method())
+            .filter_map(|method| self.routes.get(method))
+            .flatten()
+            .any(|route| paths_match(route, req))
```

Comment:
It should be possible to simplify this using the `Router::routes` method:

```rust
self.routes()
    .any(|route| route.method != req.method() && paths_match(route, req))
```

Reply:
Thanks, I'll check how that works out.